_examples/gin.go

```diff
@@ -0,0 +1,28 @@
+package examples
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+// G = global gin instance
+var G = gin.Default()
+
+// Response for /example
+func ginHandler() gin.HandlerFunc {
+	fn := func(c *gin.Context) {
+		c.JSON(200, gin.H{
+			"message": "pong",
+		})
+	}
+	return gin.HandlerFunc(fn)
+}
+
+// Init routes, bind to global G variable to be used in tests
+func init() {
+	G.GET("/example", ginHandler())
+}
+
+// Start server
+func main() {
+	G.Run()
+}
```

Comment:
All examples share the same package, so they should not define main(). Otherwise, only one example could have main. Can we construct gin instance in the test instead?

Reply:
Good point. Test works fine without it. I've removed the `main` function.